crates/bevy_ecs/src/hierarchy.rs

```diff
@@ -185,6 +185,33 @@ impl<'w> EntityWorldMut<'w> {
         self.add_related::<ChildOf>(children)
     }
 
+    /// Insert children at specific index
+    pub fn insert_children(&mut self, index: usize, children: &[Entity]) -> &mut Self {
+        let parent = self.id();
+        if children.contains(&parent) {
+            panic!("Cannot insert entity as a child of itself.");
+        }
+        if let Some(mut children_component) = self.get_mut::<Children>() {
+            children_component
+                .0
+                .retain(|value| !children.contains(value));
+            if index >= children_component.len() {
+                panic!(
+                    "Index {} out of bounds! There are only {} children!",
+                    index,
+                    children.len()
+                );
+            }
+            children_component.0.reserve(children.len());
+            let mut v = children_component.0.split_off(index);
```

Comment:
This does not update the other side of the relationship (`ChildOf`), invalidating the assumptions of the system as a whole.